### DISCUSSION TITLE
Map Scale (ie, representative fraction)

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#8845] [Q&A]
Hi all! Does deck provide a way to get the map scale and/or corresponding representative fraction based on the current zoom level?

Or… Will it need to be calculate manually? Perhaps using the WebMercatorViewport?

Thanks!

**Comment by Pessimistress** [comment]
Scale is `2 ** zoom`

**Comment by jfuehner** [comment]
@Pessimistress I *think* I am following your response.

In terms of calculating the RF from the scale (2**zoom). Assuming I have 256x256 tiles I could calculate total meters per pixels…

equatorCircumferenceMeters = 40_075_000
zoom = 0
mapWidthPixels = ( 256 * (2**zoom) ) = 256
scalePerPixel = (equatorCircumferenceMeters / mapWidthPixels) = 156_543
metersPerPixel = (1 / scalePerPixel)

RF = 1:metersPerPixel (ie, 1:156_543)

I am probably way off… Or perhaps overly complicating it.
